sqlglot/dialects/postgres.py

```diff
@@ -333,19 +334,43 @@ class Parser(parser.Parser):
 
         RANGE_PARSERS = {
             **parser.Parser.RANGE_PARSERS,
+            TokenType.AT_GT: binary_range_parser(exp.ArrayContains),
             TokenType.DAMP: binary_range_parser(exp.ArrayOverlaps),
             TokenType.DAT: lambda self, this: self.expression(
                 exp.MatchAgainst, this=self._parse_bitwise(), expressions=[this]
             ),
-            TokenType.AT_GT: binary_range_parser(exp.ArrayContains),
             TokenType.LT_AT: binary_range_parser(exp.ArrayContained),
+            TokenType.OPERATOR: lambda self, this: self._parse_operator(this),
         }
 
         STATEMENT_PARSERS = {
             **parser.Parser.STATEMENT_PARSERS,
             TokenType.END: lambda self: self._parse_commit_or_rollback(),
         }
 
+        def _parse_operator(self, this: t.Optional[exp.Expression]) -> t.Optional[exp.Expression]:
+            while True:
+                if not self._match(TokenType.L_PAREN):
+                    break
+
+                op = ""
+                while self._curr and not self._match(TokenType.R_PAREN):
+                    op += self._curr.text
```

Comment:
What do you mean?

Reply:
why are you just doing text concat instead of matching tokens?

Reply:
Because

1) It adds complexity since you have to match all possible operators (which include both single and non-single tokens, so you need to make a new collection to represent all of them and keep that collection up-to-date) and

2) I don't see the advantage of matching tokens. You could say we can parse them into an expression instead of producing a string, but what would `path.+` be parsed into? If it parsed into some known expression like `Column` or `Table`, it would be matched by a `find` or `find_all` call which is probably not the intention. So you have to make new expressions and/or add more complexity to represent it etc.